presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/util/CassandraCqlUtils.java

```diff
@@ -76,6 +76,10 @@ private static String validIdentifier(String identifier)
         if (keywords.contains(identifier.toUpperCase(ENGLISH))) {
             return quoteIdentifier(identifier);
         }
+
+        if (identifier.startsWith("_") || identifier.endsWith("_")) {
+            return quoteIdentifier(identifier);
```

Comment:
In the description you wrote

> Only alphanumeric and underscore are allowed as the identifier in Cassandra.

but the logic here is different.

Did you mean something like

```
if (!identifier.matches("[a-z]([a-z0-9_]*[a-z0-9])?") {
    return quoteIdentifier(identifier);
}
```

this would not only impl want you describe (quote everything that contains non-underscore, non-alphanumeric characters), but also

* quote names starting with digit (i'm just guessing this isn't legal either)
* quote empty names (eliminating existing check at the beginning of `validColumnName` method)
* quote non-lowercase names (eliminating existing check at the beginning of `validIdentifier` method)

(of course, such Pattern should be compiled as static field)

Reply:
Sorry, my description was insufficient. Cassandra throws an exception if an identifier contains non-underscore symbols. This case happens when Presto tries to create an table. 
So, my intent was "allowed but it is needed to be quoted in some cases". 

Exactly, my current commit misses some cases (e.g. starting/ending with digit). Let me update.

Reply:
btw, can we quote unconditionally?

i know you've considered that, but I don't know why you abandoned this idea.

Reply:
It seems like existing tests depend on the condition (I don't think the test is important), so I wanted to confirm more detail. I'll investigate it again before updating this PR.